File: [프로그래머스] 연습문제_명예의 전당1.py
def solution(k, score):
    arr = []
    answer = []
    for i in score:
        arr.append(i)   #원소 하나씩 추가
        arr.sort(reverse = True)    #내림차순 정렬
        answer.append(min(arr[:k])) #상위 k개 pick한 뒤 최솟값을 answer에 추가
    return answer

File: [프로그래머스] 연습문제_명예의 전당1.js
function solution(k, score) {
    let arr = []
    var answer = [];
    score.forEach((it, idx)=>{
        if(arr.length >= k){
            arr.sort((a, b)=>a-b)
            if(it > arr[0]){
                arr.shift();
                arr.push(it);
            }
        } else {
            arr.push(it)
        }
        arr.sort((a, b)=>a-b)
        answer.push(arr[0])
    })
    return answer;
}